php-cs-fixer

```diff
@@ -52,8 +52,14 @@ try {
     }
 }
 
+use Composer\XdebugHandler\XdebugHandler;
 use PhpCsFixer\Console\Application;
 
+// Restart if xdebug is loading, unless the environment variable PHP_CS_FIXER_ALLOW_XDEBUG is set.
+$xdebug = new XdebugHandler('PHP_CS_FIXER', '--ansi');
```

Comment:
why we need to specify colors here ?

Reply:
Because with out it the subprocess will think it is in a text only terminal and color and progress-bars won't output correctly.

Reply:
hm... how it would behave if console is missing colors indeed and we pass `--ansi` ? ?

Reply:
It only passes the parameter if the parent is in a terminal that supports color. This is not to to set the parameter but only to tell xdebughandler what the parameter is called in case it needs to use it. Other applications call it "--color" and others don't even have such a feature, so XdebugHandler can simply know the name of the parameter before hand, but again, it is only passed to the child process if relevant.